sourced/ml/transformers/basic.py

```diff
@@ -206,33 +209,45 @@ def __call__(self, head: RDD):
 
 
 class LanguageSelector(Transformer):
-    def __init__(self, languages: Union[list, tuple], blacklist=False, **kwargs):
+    def __init__(self, languages: list, blacklist=False, **kwargs):
         super().__init__(**kwargs)
         self.languages = languages
         self.blacklist = blacklist
 
     def __call__(self, files: DataFrame) -> DataFrame:
-        files = files.dropDuplicates(("blob_id",)).filter("is_binary = 'false'")
-        classified = files.classify_languages()
-        if not self.blacklist:
-            lang_filter = classified.lang == self.languages[0]
-            for lang in self.languages[1:]:
-                lang_filter |= classified.lang == lang
-        else:
-            lang_filter = classified.lang != self.languages[0]
-            for lang in self.languages[1:]:
-                lang_filter &= classified.lang != lang
-        filtered_by_lang = classified.filter(lang_filter)
-        return filtered_by_lang
+        if not isinstance(files, BlobsDataFrame):
+            self._log.warning("Argument type is not BlobsDataFrame. "
```

Comment:
We should print the warning only if `languages` is not None

Reply:
it is an error if `languages is None` here. because we do 
`return files[files.lang.isin(self.languages) != self.blacklist]`
hard to tell what we expect to happen if `self.languages is None`.
So It can not be None. We use `LanguageSelector` only in one place: https://github.com/src-d/ml/pull/284/files#diff-370750368c8314a487dfbae4ca854901R347 and with `.maybe()`. it checks` if languages is None`

Reply:
Then you must type assert for `BlobsDataFrame` (or `BlobsWithLanguageDataFrame`?) instead of the warning. The input type must be exactly one possible class and it is the responsibility of the caller to ensure that.